modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavascriptClientCodegen.java

```diff
@@ -795,6 +795,11 @@ public CodegenModel fromModel(String name, Model model, Map<String, Model> allDe
             if (mm.getAdditionalProperties() != null) {
                 codegenModel.vendorExtensions.put("x-isMap", true);
                 codegenModel.vendorExtensions.put("x-itemType", getSwaggerType(mm.getAdditionalProperties()));
+            } else {
+                String type = mm.getType();
+                if (type != null && (type.equals("number") || type.equals("integer") || type.equals("string") || type.equals("boolean") || type.equals("null"))) {
```

Comment:
To reduce cyclomatic complexity, could this line perhaps be replace with something like:
`String[] PRIMITIVE_TYPES = new String[] {"number", "integer","string","boolean", "null"};`
`Arrays.asList(PRIMITIVE_TYPES).contains(type)`

Reply:
Fixed. It also improves readability.